[utfcpp] Build error

**Host Environment**
- OS: Windows
- Compiler: g++ (Rev3, Built by MSYS2 project) 12.1.0
-  vcpkg-tool version: 2022-06-17-9268e366206712e38102b28dbd1617697a99ff2e
-  vcpkg-scripts version: 68b7fec22 2022-07-15

**To Reproduce**
Steps to reproduce the behavior:
Configure vcpkg for MinGW as described [here ](https://github.com/microsoft/vcpkg/blob/master/docs/users/mingw.md)
`./vcpkg install ogre`

**Failure logs**
```
Installing 9/31 utfcpp:x64-mingw-dynamic...
Building utfcpp[core]:x64-mingw-dynamic...
warning: -- Using community triplet x64-mingw-dynamic. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: <snip>\vcpkg\triplets\community\x64-mingw-dynamic.cmake
-- Downloading https://github.com/nemtrif/utfcpp/archive/v3.2.1.tar.gz -> nemtrif-utfcpp-v3.2.1.tar.gz...
-- Extracting source<snip>/vcpkg/downloads/nemtrif-utfcpp-v3.2.1.tar.gz
-- Using source at <snip>/buildtrees/utfcpp/src/v3.2.1-9f3df3227a.clean
-- Configuring x64-mingw-dynamic
-- Building x64-mingw-dynamic-dbg
-- Building x64-mingw-dynamic-rel
CMake Error at scripts/cmake/vcpkg_fixup_cmake_targets.cmake:36 (message):
  '<snip>/vcpkg/packages/utfcpp_x64-mingw-dynamic/debug/cmake'
  does not exist.
Call Stack (most recent call first):
  ports/utfcpp/portfile.cmake:21 (vcpkg_fixup_cmake_targets)
  scripts/ports.cmake:147 (include)
```

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

--- a/ports/utfcpp/portfile.cmake
+++ b/ports/utfcpp/portfile.cmake
@@ -16,7 +16,7 @@ vcpkg_cmake_configure(
 
 vcpkg_cmake_install()
 
-if (VCPKG_TARGET_IS_WINDOWS)
+if (VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
     vcpkg_cmake_config_fixup(PACKAGE_NAME utf8cpp CONFIG_PATH cmake)
 else()
     vcpkg_cmake_config_fixup(PACKAGE_NAME utf8cpp CONFIG_PATH lib/cmake/utf8cpp)

--- a/ports/utfcpp/portfile.cmake
+++ b/ports/utfcpp/portfile.cmake
@@ -6,24 +6,23 @@ vcpkg_from_github(
     HEAD_REF master
 )
 
-vcpkg_configure_cmake(
-    SOURCE_PATH ${SOURCE_PATH}
-    PREFER_NINJA
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
     OPTIONS
         -DUTF8_INSTALL=ON
         -DUTF8_SAMPLES=OFF
         -DUTF8_TESTS=OFF
 )
 
-vcpkg_install_cmake()
+vcpkg_cmake_install()
 
 if (VCPKG_TARGET_IS_WINDOWS)
-    vcpkg_fixup_cmake_targets(CONFIG_PATH cmake TARGET_PATH share/utf8cpp)
+    vcpkg_cmake_config_fixup(PACKAGE_NAME utf8cpp CONFIG_PATH cmake)
 else()
-    vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/utf8cpp TARGET_PATH share/utf8cpp)
+    vcpkg_cmake_config_fixup(PACKAGE_NAME utf8cpp CONFIG_PATH lib/cmake/utf8cpp)
 endif()
 
 # Header only
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug ${CURRENT_PACKAGES_DIR}/lib)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug" "${CURRENT_PACKAGES_DIR}/lib")
 
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

--- a/ports/utfcpp/vcpkg.json
+++ b/ports/utfcpp/vcpkg.json
@@ -1,6 +1,18 @@
 {
   "name": "utfcpp",
   "version": "3.2.1",
+  "port-version": 1,
   "description": "UTF-8 with C++ in a Portable Way",
-  "homepage": "https://github.com/nemtrif/utfcpp"
+  "homepage": "https://github.com/nemtrif/utfcpp",
+  "license": "BSL-1.0",
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ]
 }

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -7546,7 +7546,7 @@
     },
     "utfcpp": {
       "baseline": "3.2.1",
-      "port-version": 0
+      "port-version": 1
     },
     "utfz": {
       "baseline": "1.2",

--- a/versions/u-/utfcpp.json
+++ b/versions/u-/utfcpp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c47a0911020f386e46e4f920a55fec656976d2ef",
+      "version": "3.2.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "27a1e587de353ed0d9559d5e4627ed2811960c07",
       "version": "3.2.1",
       "port-version": 0

